src/Engine/TranskribusClient.php

```diff
@@ -137,8 +146,97 @@ private function request(
                 return $content;
             }
             $this->throwException(0);
-        } else {
-            $this->throwException($statusCode);
         }
+
+        if (401 === $statusCode) {
+            $this->setAccessToken();
+
+            $body['headers']['Authorization'] = 'Bearer '.$this->accessToken;
+            $response = $this->httpClient->request($method, $url, $body);
+            $statusCode = $response->getStatusCode();
+
+            if (200 === $statusCode) {
+                $content = json_decode($response->getContent());
+
+                if (!empty($content)) {
+                    return $content;
+                }
+                $this->throwException(0);
+            }
+        }
+        $this->throwException($statusCode);
+    }
+
+    private function setAccessToken(): void
+    {
+        $response = self::getRefreshTokenResponse($this->refreshToken, $this->httpClient);
+        $statusCode = $response->getStatusCode();
+        if (200 === $statusCode) {
+            $content = json_decode($response->getContent());
+
+            if (!empty($content)) {
+                $this->accessToken = $content->{'access_token'};
+            }
+            $this->throwException(0);
+        }
+        $this->throwException($statusCode);
+    }
+
+    /**
+     * @param string $token
+     * @param HttpClientInterface $client
+     */
+    public static function getRefreshTokenResponse(string $token, HttpClientInterface $client): object
+    {
+        $body = [
+            'grant_type' => 'refresh_token',
+            'client_id' => 'processing-api-client',
+            'refresh_token' => $token,
+        ];
+
+        $response = self::authRequest($body, $client);
+        return $response;
+    }
+
+    /**
+     * @param string $userName
+     * @param string $password
+     * @param HttpClientInterface $client
+     */
+    public static function getAccessTokenResponse(
+        string $userName,
+        string $password,
+        HttpClientInterface $client
+    ): object {
+        $body = [
```

Comment:
I believe this function is not being called anywhere.

Reply:
The Intent is for it to be used in the TranskribusAuthCommand class to have the requests in one place